apps/web/lib/hooks/useSuperfetch.ts

```diff
@@ -0,0 +1,38 @@
+import { useSession } from "next-auth/react";
+
+interface SuperOptions extends Omit<Omit<RequestInit, "headers">, "body"> {
+  body: object;
+}
+
+export default function useSuperfetch(
+  url: string,
+  //disable ts suggestion for headers since we are setting them up manually and allow body to be object
+  options?: SuperOptions
+) {
+  const { data: session } = useSession();
+
+  const superfetch = () => {
```

Comment:
```suggestion
  const superfetch = (
    url: string,
    //disable ts suggestion for headers since we are setting them up manually and allow body to be object
    options?: SuperOptions
  ) => {
```

What do you think about making it more flexible (more super 😮) by passing these parameters inside superfetch?

Reply:
By passing them like this, we are marking `url` and `options` passed to hook as unused. And fetch won't work that way. 😞 

![image](https://github.com/intive/patronage2023-js/assets/79754698/aea0f818-1d6a-4be2-b978-02d3455aae20)

Reply:
I'm talking about moving these attributes from the hook to the superfetch below 

Basically, it would look like this
```tsx
export default function useSuperfetch() {
  const { data: session } = useSession();

  const superfetch = (
    url: string,
    //disable ts suggestion for headers since we are setting them up manually and allow body to be object
    options?: SuperOptions
  ) => {
```

and then the superfetch call changes slightly

```tsx
  const fetch = useSuperfetch();
  const URL = ``;
  const options = {};
  fetch(URL, options);
```
The main goal is preserved because you insert the token inside. The benefit I see in this approach is that we don't stick to the URL and option declarations in the functional component. This can be useful in a dynamically changing option body, for example in a tanstack queryFn

Reply:
You are right that would be more super to have and keep hook parameters empty.